cmd/rtmp_segment_packer/main.go

```diff
@@ -0,0 +1,344 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"io"
+	"strings"
+
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/kz26/m3u8"
+	"github.com/livepeer/lpms"
+	"github.com/livepeer/lpms/stream"
+	"github.com/nareix/joy4/av"
+)
+
+//This is basically a test for segmenting videos.  After much research, the current best approach is to use the segment function inside FFMpeg.
+//However, this test serves as a document so we can revisit later.
+
+var StagedStream SegmentStream
+var transcodeCount int
+
+type StreamDB struct {
+	db map[string]stream.Stream
+}
+
+type Segment struct {
+	availSpace time.Duration
+	packets    []av.Packet
+}
+
+func NewSegment() *Segment {
+	return &Segment{time.Second * 2, make([]av.Packet, 0, 100)}
+}
+
+type SegmentStream struct {
+	StreamID    string
+	Headers     []av.CodecData
+	Segments    []*Segment
+	RTMPTimeout time.Duration
+}
+
+func NewSegmentStream(streamID string) *SegmentStream {
+	return &SegmentStream{streamID, nil, make([]*Segment, 0, 100), time.Second * 10}
+}
+
+func (s *SegmentStream) GetStreamID() string {
+	return s.StreamID
+}
+
+func (s *SegmentStream) Len() int64 {
+	return int64(len(s.Segments))
+}
+
+func (s *SegmentStream) ReadRTMPFromStream(ctx context.Context, dst av.MuxCloser) error {
+	defer dst.Close()
+
+	dst.WriteHeader(s.Headers)
+	transcodeCount = transcodeCount + 1
+
+	if transcodeCount%2 == 0 {
+		glog.Infof("Writing seg 2")
+		for _, p := range s.Segments[2].packets {
+			err := dst.WritePacket(p)
+			time.Sleep(time.Millisecond * 5)
+
+			if err != nil {
+				glog.Infof("Error writing RTMP packet from Stream %v to mux", s.StreamID)
+				return err
+			}
+		}
+		glog.Infof("Writing seg 3")
+		for _, p := range s.Segments[3].packets {
+			err := dst.WritePacket(p)
+			time.Sleep(time.Millisecond * 5)
+
+			if err != nil {
+				glog.Infof("Error writing RTMP packet from Stream %v to mux", s.StreamID)
+				dst.WriteTrailer()
+				return err
+			}
+		}
+
+		glog.Infof("Writing seg 6")
```

Comment:
This is just a test.  I was trying to examine individual segments and see if transcoding output would be consistent if you skipped certain segments - and turns out no.  This made me pursue an alternative approach where we use FFMpeg to segment the video first.